### PR TITLE
Send temperatures

### PR DESCRIPTION
**Pull request by derpicknicker1**
Added the ability to publish temperature changes and options to deactivate subtopics
See #14

### CODE REVIEW

File: octoprint_mqtt/__init__.py [hunk 1]
```diff
@@ -23,7 +23,14 @@ def __init__(self):
 		self._mqtt_publish_queue = deque()
 		self._mqtt_subscribe_queue = deque()
 
+		self.lastTemp = {}
+
+		
+
 	def initialize(self):
+		self.printer_callback = octoprint.printer.PrinterCallback()
```
Comment (Xasin) [inline octoprint_mqtt/__init__.py]: How come you have to register the function on the callback like this? 
The rest of the functions of this class must use some other form of callback too, right?

I'm not an expert though, so maybe the other functions have another way of being called.

Comment (derpicknicker1) [inline octoprint_mqtt/__init__.py]: I c&p from here: https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/blob/master/octoprint_firmwareupdater/__init__.py#L259

Comment (Xasin) [inline octoprint_mqtt/__init__.py]: I was a bit confused at first why this callback had to be done differently... 
But it seems like the Octoprint API doesn't supply a fitting Mixin type for temperature status reports! 

Carry on >.>

Comment (foosel) [inline octoprint_mqtt/__init__.py]: Actually, having your plugin just subclass `PrinterCallback` and then doing `self._printer.register_callback(self)` should work just fine as well and would be cleaner ;)

As a side note, that FirmwareUpdater plugin was written by a former colleague and needs some love here and there, so sorry for the "bad example" it set :)

**Comment by foosel** [comment]
Thanks for that! Will try to get around to do a proper look ASAP. Sorry in advance for any delays in that!

File: octoprint_mqtt/__init__.py (hunk 1 above)
Comment (derpicknicker1) [inline octoprint_mqtt/__init__.py]: At first try i did it this way (giving the plugin the subclass `PrinterCallback`), but i forgot to register the callback. I fixed it by new request.

**Comment by foosel** [comment]
I've merged this to the `devel` branch. I'll give it some more testing and then release 0.5 with it. Thanks!

**Comment by foosel** [comment]
PS: PR should automatically be marked as closed once the commits hit `master`.

**Pull request merged**